### PR TITLE
Use where syntax in umfpack.jl

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

**Review by andreasnoack — APPROVED** [review]

**Review by Sacha0 — APPROVED** [review]
Thanks @musm! :) (Absent objections, requests for time, or someone beating me to it, I plan to merge this tomorrow morning.)

**Comment by tkelman** [comment]
will conflict with https://github.com/JuliaLang/julia/pull/21889, which should be higher priority

**Comment by Sacha0** [comment]
Rebase with #21889 in? :)

**Review by Sacha0 — APPROVED** [review]
lgtm for the rebased version! :) (Absent objections, requests for time, or someone beating me to it, I plan to merge this later today.)

File: base/sparse/umfpack.jl [hunk 1]
```diff
@@ -382,24 +382,24 @@ function nnz(lu::UmfpackLU)
...
-A_ldiv_B!{T<:UMFVTypes}(X::StridedVecOrMat{T}, lu::UmfpackLU{T}, B::StridedVecOrMat{T}) =
+A_ldiv_B!(X::StridedVecOrMat{T}, lu::UmfpackLU{T}, B::StridedVecOrMat{T}) where {T<:UMFVTypes} =
     _Aq_ldiv_B!(X, lu, B, UMFPACK_A)
-At_ldiv_B!{T<:UMFVTypes}(X::StridedVecOrMat{T}, lu::UmfpackLU{T}, B::StridedVecOrMat{T}) =
+At_ldiv_B!(X::StridedVecOrMat{T}, lu::UmfpackLU{T}, B::StridedVecOrMat{T}) where {T<:UMFVTypes} =
     _Aq_ldiv_B!(X, lu, B, UMFPACK_Aat)
-Ac_ldiv_B!{T<:UMFVTypes}(X::StridedVecOrMat{T}, lu::UmfpackLU{T}, B::StridedVecOrMat{T}) =
+Ac_ldiv_B!(X::StridedVecOrMat{T}, lu::UmfpackLU{T}, B::StridedVecOrMat{T}) where {T<:UMFVTypes} =
     _Aq_ldiv_B!(X, lu, B, UMFPACK_At)
 A_ldiv_B!{Tb<:Complex}(X::StridedVecOrMat{Tb}, lu::UmfpackLU{Float64}, B::StridedVecOrMat{Tb}) =
```
Comment (tkelman) [inline base/sparse/umfpack.jl]: get this one too?

Comment (musm) [inline base/sparse/umfpack.jl]: thanks

**Comment by Sacha0** [comment]
Gah, should've squashed. Apologies!

**Pull request merged**